src/util.cc

```diff
@@ -404,19 +396,35 @@ void seed_random(unsigned int seed)
 		srandom(seed);
 	}
 
-void init_random_seed(const char* read_file, const char* write_file, bool use_empty_seeds)
+void init_random_seed(const char* read_file, const char* write_file, bool use_empty_seeds,
+                      std::string seed_string)
```

Comment:
Either pass this as const std::string& or turn it into a std::string_view.

Reply:
Thanks - fmemopen wants a `void *` so either of those would require a `const_cast` then.

I see that `fmemopen()` isn't available on window, so I'll see about switching this to input streams instead.

Reply:
Just two random thoughts: 1) The function has become quite cluttered. How about separating the file-related stuff and the seed generation (e.g., by extracting the latter to its own function). 2) Maybe just setting the appropriate `seed_string` to get rid of the `use_empty_seeds` special case would improve readability.

Reply:
I can see, but I also didn't want to re-structure all that much in code where my IDE shows half the things ifdef'ed out :fearful:  While `init_random_seed()` is in zeek::detail, not breaking API compat if we don't need to was a bit in mind, too. Unless Tim thinks this has now crossed some threshold I'll keep it as is for now.